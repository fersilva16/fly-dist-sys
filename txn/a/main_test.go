package main

import (
	"gossip-gloomers/testutils"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	maelstrom "github.com/jepsen-io/maelstrom/demo/go"
	"github.com/stretchr/testify/require"
)

func TestEcho(t *testing.T) {
	require := require.New(t)

	node = maelstrom.NewNode()

	link := testutils.NewLink(node)
	client := testutils.NewClient("c0", link)

	go main()

	err := client.InitNode("n0", []string{"n0"})

	require.NoError(err)

	txnOutput1, err := client.RPC(TxnRequest{
		MessageBody: maelstrom.MessageBody{
			Type:  "txn",
			MsgID: 2,
		},

		Txn: Txn{
			{WRITE, 1, 1},
			{READ, 2, nil},
		},
	})

	require.NoError(err)

	snaps.MatchSnapshot(t, txnOutput1)

	txnOutput2, err := client.RPC(TxnRequest{
		MessageBody: maelstrom.MessageBody{
			Type:  "txn",
			MsgID: 3,
		},

		Txn: Txn{
			{READ, 1, nil},
			{WRITE, 2, 2},
		},
	})

	require.NoError(err)

	snaps.MatchSnapshot(t, txnOutput2)
}
