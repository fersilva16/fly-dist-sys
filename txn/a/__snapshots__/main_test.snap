
[TestEcho - 1]
{"src":"n0","dest":"c0","body":{"in_reply_to":2,"txn":[["w",1,1],["r",2,0]],"type":"txn_ok"}}
---

[TestEcho - 2]
{"src":"n0","dest":"c0","body":{"in_reply_to":3,"txn":[["r",1,1],["w",2,2]],"type":"txn_ok"}}
---
